app/src/main/java/com/example/beni/myapplication/LoginActivity.java

```diff
@@ -0,0 +1,112 @@
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Header;
+
+public class LoginActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+//        findViewById(R.id.layout_login).setOnTouchListener(new View.OnTouchListener() {
+//            @Override
+//            public boolean onTouch(View v, MotionEvent event) {
+//                closeKeyboard(v);
+//                return false;
+//            }
+//        });
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        if(preferences.getString("auth",null)!= null){
+            Intent intent = new Intent(this, ProfileActivity.class);
+            startActivity(intent);
+            finish();
+        }
+        Button b = (Button)findViewById(R.id.button);
+
+        b.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                EditText t = (EditText)findViewById(R.id.pass);
+                final String pass =t.getText().toString();
+                t = (EditText)findViewById(R.id.user);
+                final String user = t.getText().toString();
+                final String authHash = Credentials.basic(user,pass);
+                Call<LoginData> callable = GitHub.Service.get().checkAuth(authHash);
+
+                callable.enqueue(new Callback<LoginData>() {
+                    @Override
+                    public void onResponse(Call<LoginData> call, Response<LoginData> response) {
+
+                        if(response.isSuccessful()){
+                        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(LoginActivity.this);
+                        preferences.edit().putString("auth",authHash)
+                                .putString("username",user)
+                                .apply();
+                        Intent intent = new Intent(LoginActivity.this, ProfileActivity.class);
+//                    EditText editText = (EditText) findViewById(R.id.editText);
+//                    String message = editText.getText().toString();
+//                    intent.putExtra(EXTRA_MESSAGE, message);
+                        startActivity(intent);}
+                        else{
+                            Context context = getApplicationContext();
+                            CharSequence text = "Wrong password or username!";
+                            int duration = Toast.LENGTH_SHORT;
+
+                            Toast toast = Toast.makeText(context, text, duration);
+                            toast.show();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<LoginData> call, Throwable t) {
+                        Context context = getApplicationContext();
```

Comment:
onFailure is not called for invalid password/username. It's called for other reasons. Use the t throwable to determine whether it was one of the following cases:
1. timeout (t is an instance of TimeoutException)
2. no internet connection (t is an instance of UnknownHostException)